core/src/main/java/org/elasticsearch/index/mapper/TextFieldMapper.java

```diff
@@ -285,9 +285,10 @@ public Query nullValueQuery() {
         @Override
         public IndexFieldData.Builder fielddataBuilder() {
             if (fielddata == false) {
-                throw new IllegalArgumentException("Fielddata is disabled on text fields by default. Set fielddata=true on [" + name()
-                        + "] in order to load fielddata in memory by uninverting the inverted index. Note that this can however "
-                        + "use significant memory.");
+                throw new IllegalArgumentException("Fielddata is disabled on text fields by default. For sorting, aggregations, or "
+                        + "accessing field values in scripts, either: 1. Use the [" + name() + ".keyword] field instead; or 2. Set "
```

Comment:
I'd say "accessing *text* field values in scripts".

The advice to use the `.keyword` field only makes sense if the `.keyword` field exists. I wouldn't include it unless you've checked that it exists. Which I think is a good idea. If you throw a subclass of IllegalArgumentException and caught it at `QueryShardContext#getForField` you could probably check for the subfield and talk about it. That is a much more complex change then this is, but I think it is the only way I'd be ok with including the `.keyword` advice. In that case I'd just look for any subfield that is a `keyword` type.